src/app/components/shared/datetime/datetime/datetime.component.spec.ts

```diff
@@ -0,0 +1,230 @@
+    // for this test, we create th DateTime object in UTC+02:00
+    // we should therefore see the hours increase by 6
+    name: "should display the correct dateTime if the Luxon dateTime object is a different utc offset to the users local timezone",
+    // by using +02:00, we know the offset, but not the timezone
+    // however, because this value will be localized to the users local timezone, we should see the timezone emitted
+    // in the users local timezone
+    value: DateTime.fromISO("2020-01-01T12:10:11.000+02:00"),
+    expectedText: "2020-01-01 18:10:11",
+    // notice that the tooltips are still in the test runners mock timezone
+    // this is because the emitted date/time has been localized to the test runners timezone
+    expectedTooltip: "2020-01-01 18:10:11.000 Australia/Perth +08:00",
+    expectedDateTimeAttribute: "2020-01-01T18:10:11.000+08:00",
+  },
+  {
+    // this test is the same as the previous offset test, except that we know both the timezone and offset
+    // in this test. (as opposed to the previous test where we only knew the offset)
+    name: "should display the correct DateTime if the Luxon dateTime object is a different timezone to the users local timezone",
+    // by using Australia/Sydney, we know the timezone and offset
+    value: DateTime.fromISO("2020-01-01T12:10:11.000Z").setZone(
+      "Australia/Sydney",
+      { keepLocalTime: true }
+    ),
+    expectedText: "2020-01-01 09:10:11",
+    expectedTooltip: "2020-01-01 09:10:11.000 Australia/Perth +08:00",
+    expectedDateTimeAttribute: "2020-01-01T09:10:11.000+08:00",
+  },
+  {
+    name: "should increment date if timezone localization causes the date to increment",
+    value: DateTime.fromISO("2020-01-01T23:10:11.000Z"),
+    expectedText: "2020-01-02 07:10:11",
+    expectedTooltip: "2020-01-02 07:10:11.000 Australia/Perth +08:00",
+    expectedDateTimeAttribute: "2020-01-02T07:10:11.000+08:00",
+  },
+  {
+    name: "should only display the date if dateOnly is set",
+    value: DateTime.fromISO("2020-01-01T12:10:11.000Z"),
+    expectedText: "2020-01-01",
+    expectedTooltip: "2020-01-01 20:10:11.000 Australia/Perth +08:00",
+    dateOnly: true,
+    expectedDateTimeAttribute: "2020-01-01T20:10:11.000+08:00",
+  },
+  {
+    name: "should only display the time if timeOnly is set",
+    value: DateTime.fromISO("2020-01-01T12:10:11.000Z"),
+    expectedText: "20:10:11",
+    expectedTooltip: "2020-01-01 20:10:11.000 Australia/Perth +08:00",
+    timeOnly: true,
+    expectedDateTimeAttribute: "2020-01-01T20:10:11.000+08:00",
+  },
+  {
+    name: "should display full datetime if both dateOnly and timeOnly are set",
+    value: DateTime.fromISO("2020-01-01T12:10:11.000Z"),
+    expectedText: "2020-01-01 20:10:11",
+    expectedTooltip: "2020-01-01 20:10:11.000 Australia/Perth +08:00",
+    dateOnly: true,
+    timeOnly: true,
+    expectedDateTimeAttribute: "2020-01-01T20:10:11.000+08:00",
+  },
+  {
```

Comment:
Throwing a runtime console error works for me.

My original justification as to why we might want both was:

Imagine we have a place in the website where we want to conditionally show date, time or both date & time.  
We should be able to do something like `<baw-datetime [timeOnly]="true" [dateOnly]="true" />` (where `true` is set reactively by view model values)

Reply:
Nah I'm really suggesting changing the design spec semantics of these components. (changing the names)

- if no modifier attributes are set then show standard format
- if `date` is set show the date
- if `time` is set show the time
- if `<baw-datetime value=... date time seconds zone>` then show all of those components

Proposed change allows for later extensions (`zone`, `seconds`) down the line without reading non-sensically (`<baw-datetime value=... dateOnly timeOnly secondsOnly zoneOnly>`) and still allows for binding (and it even makes more sense since the booleans aren't negated)

Reply:
Done